eclair-core/src/main/scala/fr/acinq/eclair/channel/InteractiveTxBuilder.scala

```diff
@@ -763,8 +763,9 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
       // When both peers contribute the same amount, the peer with the lowest pubkey must transmit its `tx_signatures` first.
       LexicographicalOrdering.isLessThan(commitments.localParams.nodeId.value, commitments.remoteNodeId.value)
     } else {
-      // The peer with the lowest total of input amount must transmit its `tx_signatures` first.
-      completeTx.localAmountIn < completeTx.remoteAmountIn
+      // If our peer didn't contribute to the transaction, we don't need to wait for their `tx_signatures`.
+      // Otherwise, the peer with the lowest total of input amount must transmit its `tx_signatures` first.
+      completeTx.remoteAmountIn == 0.sat || completeTx.localAmountIn < completeTx.remoteAmountIn
```

Comment:
Nit: I think clarity beats conciseness in this instance:

```scala
    val shouldSignFirst = if (completeTx.localAmountIn == completeTx.remoteAmountIn) {
      // When both peers contribute the same amount, the peer with the lowest pubkey must transmit its `tx_signatures` first.
      LexicographicalOrdering.isLessThan(commitments.localParams.nodeId.value, commitments.remoteNodeId.value)
    } else if (completeTx.remoteAmountIn == 0.sat) {
      // If our peer didn't contribute to the transaction, we don't need to wait for their `tx_signatures`, they will be
      // empty anyway.
      true
    } else {
      // Otherwise, the peer with the lowest total of input amount must transmit its `tx_signatures` first.
      completeTx.localAmountIn < completeTx.remoteAmountIn
    }
```

Reply:
Heh, that was actually the first version of the code I wrote, done in https://github.com/ACINQ/eclair/pull/2560/commits/3d0c4297c6760db27f722b7d1f83cda483568d14